exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/transactions/TransactionPreconditions.java

```diff
@@ -31,31 +31,31 @@ private TransactionPreconditions() {
   }
 
   @CanIgnoreReturnValue
-  static <MessageT extends Message> MessageT checkTransaction(
-      MessageT message, short expectedTxId) {
-    checkServiceId(message);
-    checkTransactionId(message, expectedTxId);
-    return message;
+  static <T extends RawTransaction> T checkTransaction(T transaction, short expectedTxId) {
```

Comment:
It used to work with BinaryMessage & just Message, returning exactly what you pass, so you can do
```java
consumeMessage(checkTransaction(message, 10)) // works with consumeMessage(BinaryMessage) & consumeMessage(Message)
```

Please simplify to checkTransaction(RawTransaction, short), because with RawTransaction it does not seem to be needed (especially in void methods below).